### DISCUSSION TITLE
Disable integration with Next i18n routing

### DISCUSSION THREAD

**Discussion by abarani** [vinissimus/next-translate#550] [Q&A]
Hey, is it possible to use next-translate without integrating with Next.js i18n routing and rely only on `Accept-Language` header and cookie only?

From what I understand adding a locale to the `locales` option in `i18n.js` config automatically adds it to the Next.js config and thus enables sub-path routing for the app. What I'm trying to achieve is keeping only a single path for each page but display it localized based solely on `Accept-Language` header or cookie (if set). This is for an SSR only app.

Great project by the way!

**Comment by aralroca** [comment]
@abarani since version 0.19 all the routing part was moved to Next.js core. So right now next-translate by itself has no way to handle routing. Before version 0.19, there was a "build step" to generate the "pages" directory with all the languages.

The change to 0.19 has greatly simplified the library making everything easier, simpler and more comfortable to use. What is the reason to don't want to use the i18n routing of Next.js? I would like to know.

**Reply by abarani** [reply]
I'm developing a small app that handles user authentication. It integrates with an external service that redirects to the various pages of the app based on a static list of URLs.

So with the default Next.js i18n routing I'm only able to land the user to the default locale URL of a page but I can't take advantage of the automatic locale detection (as far as I know [this is only done on the root page on Next.js](https://nextjs.org/docs/advanced-features/i18n-routing#automatic-locale-detection)).

As a simple solution I was thinking on manually detecting the locale from the `Accept-Language` header and completely ignore the routing as it is not something I care of in this app.

Does it make sense?

**Reply by aralroca** [reply]
You can use `localeDetection: false` inside the `i18n.js` file. Documentation: https://nextjs.org/docs/advanced-features/i18n-routing#disabling-automatic-locale-detection maybe it helps
